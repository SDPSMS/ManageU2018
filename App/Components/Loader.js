import React from 'react'
import PropTypes from 'prop-types'
import { ActivityIndicator, View } from 'react-native'

/**
 * Component to create a loader (indicating that something is working on the background
 * @param size determine the size of loader
 * @returns {*}
 * @constructor
 */
// size refer to the activity indicator's size.
const Loader = ({ size }) => {
  return (
    <View styles={styles.loader}>
      <ActivityIndicator
        size={size || 'small'}
        color={'#0000ff'}
      />
    </View>
  )
}

export default Loader

Loader.propTypes = {
  size: PropTypes.string
}

const styles = {
  // TODO: make it to the middle. (center)
  loader: {
    justifyContent: 'center',
    alignItems: 'center'
  }
}
