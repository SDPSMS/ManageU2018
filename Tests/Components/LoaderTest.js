import 'react-native'
import React from 'react'
import Loader from '../../App/Components/Loader'

import renderer from 'react-test-renderer'

test('Loader component renders correcty', () => {
  const tree = renderer.create(<Loader />).toJSON()
  expect(tree).toMatchSnapshot()
})
