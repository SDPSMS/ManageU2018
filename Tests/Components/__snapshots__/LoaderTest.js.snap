// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component renders correcty 1`] = `
<View
  styles={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
